test/python/transpiler/test_pass_scheduler.py

```diff
@@ -550,6 +550,158 @@ def test_fresh_initial_state(self):
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 3",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 3 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 2",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 2 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 2",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 2 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 2",
+        ]
+        self.assertScheduler(self.circuit, self.passmanager, expected)
+
+    def test_multiple_nested_conditionals(self):
+        """Run multiple nested conditionals."""
+        nested_conditional1 = [
+            ConditionalController(
+                [PassB_TP_RA_PA()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+        nested_conditional2 = [
+            ConditionalController(
+                [PassC_TP_RA_PA()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+        nested_conditional3 = [
+            ConditionalController(
+                [PassD_TP_NR_NP()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+
+        self.passmanager.append(
+            [PassE_AP_NR_NP(True)]
+            + nested_conditional1
+            + [PassE_AP_NR_NP(False)]
+            + nested_conditional2
+            + [PassE_AP_NR_NP(True)]
+            + nested_conditional3
+        )
+
+        expected = [
+            "run analysis pass PassE_AP_NR_NP",
+            "set property as True",
+            "run transformation pass PassA_TP_NR_NP",
+            "run transformation pass PassB_TP_RA_PA",
+            "run analysis pass PassE_AP_NR_NP",
+            "set property as False",
+            "run analysis pass PassE_AP_NR_NP",
+            "set property as True",
+            "run transformation pass PassD_TP_NR_NP",
+            "argument None",
+        ]
+
+        self.assertScheduler(self.circuit, self.passmanager, expected)
+
+    def test_nested_do_while(self):
+        """Run nested do while."""
+        nested_do_while = [
+            DoWhileController(
+                [PassK_check_fixed_point_property(), PassF_reduce_dag_property()],
+                do_while=lambda property_set: not property_set["property_fixed_point"],
+            )
+        ]
+
+        self.passmanager.append([PassD_TP_NR_NP()] + nested_do_while + [PassD_TP_NR_NP()])
```

Comment:
Again, this does not test a feature introduced in this PR. There is only a single conditional here. The PR allows nesting two or more conditionals.

Reply:
I have removed this test in a commit.